### PR TITLE
[WIP] feat: Azure adapter

### PR DESCRIPTION
**Pull request by mcansh**
feat: initial azure adapter based on https://github.com/danielgary/remix-azure-template

example repo: https://github.com/mcansh/remix-on-azure
example deployment: http://brave-wave-0feb78710.azurestaticapps.net

Signed-off-by: Logan McAnsh <logan@mcan.sh>

TODO: 
- [x] add tests for createRequestHandler 
- [x] add tests for createRemixRequest
- [x] figure out why `set-cookie` is a string instead of an array of strings
- [x] fix dev mode..
- [ ] wait for azure swa to support node 14

holding off on merging this, there is a critical issue with dev mode and it not properly rewriting urls, the req.url is that of the function we're rewriting to, not the original url

### CODE REVIEW

**Comment by michaelwarren1106** [comment]
im interested in this template for sure. About to make a remix app and want to deploy on an Azure Static Site. Any idea if this will get picked up again?

**Comment by danielgary** [comment]
This should be ready to be merged soon.  Azure SWA Node 14/16 support is available in Western Europe and rolling out everywhere else this week.  Here's a demo running on Azure SWA: https://nice-wave-0a55b3c03.1.azurestaticapps.net/

**Comment by IvanFLeon** [comment]
This is ready to be merged
Per the Azure SWA documentation: https://docs.microsoft.com/en-us/azure/static-web-apps/apis

Node.js 14 and Node.js 16(Preview) are now supported as api runtimes.

File: packages/remix-azure/server.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  build: ServerBuild;
+  getLoadContext?: GetLoadContextFunction;
+  mode?: string;
+}): AzureFunction {
+  let platform: ServerPlatform = { formatServerError };
+  let handleRequest = createRemixRequestHandler(build, platform, mode);
+
+  return async (_context: Context, req: HttpRequest) => {
+    let request = createRemixRequest(req);
+    let loadContext = getLoadContext ? getLoadContext(req) : undefined;
```
Comment (IvanFLeon) [inline packages/remix-azure/server.ts:49]: This could be simplified as `getLoadContext?.(req)`

**Pull request closed without merging**